pkg/subctl/cmd/validate_deployment.go

```diff
@@ -140,3 +145,32 @@ func CheckDaemonset(k8sClient kubernetes.Interface, namespace, daemonSetName str
 
 	return true
 }
+
+func checkPodsStatus(k8sClient kubernetes.Interface, operatorNamespace string) bool {
+	pods, err := k8sClient.CoreV1().Pods(operatorNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		message := fmt.Sprintf("Error obtaining Pods list: %v", err)
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+		return false
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != v1.PodRunning {
+			message := fmt.Sprintf("Pod %q is not running. (current state is %v)", pod.Name, pod.Status.Phase)
+			status.QueueFailureMessage(message)
+			status.End(cli.Failure)
+			return false
+		}
+
+		for _, c := range pod.Status.ContainerStatuses {
+			if c.RestartCount >= 5 {
+				message := fmt.Sprintf("Pod %q has restarted %d times", pod.Name, c.RestartCount)
+				status.QueueWarningMessage(message)
+				status.End(cli.Success)
```

Comment:
I think you should move this `status.End(cli.Success)` to Line 174

Reply:
Oh sorry, I clicked rereview again, you were too quick @maayanf24  ^

Reply:
I removed `status.End(cli.Success)` from here since it's not needed.
The return value of this func will be true for this case, and then at line 99 there will be a call for `status.End(cli.Success)`